### ISSUE TITLE
stream: support more options

### ISSUE THREAD

**Issue by legendecas** [Rokid/ShadowNode#185]
Support following construction options for stream

- [ ] objectMode
- [ ] decodeStrings

**Comment by Txiaozhe** [comment]
Can you assign it to me? I will try! Thanks:)

**Comment by legendecas** [comment]
> Can you assign it to me? I will try! Thanks:)

Yes, definitely

**Comment by Txiaozhe** [comment]
Thanks!:)

**Comment by yorkie** [comment]
@Txiaozhe we could discuss here, the object mode is designed for writing stuffs in buffer/string, and create pipelines for object processing. Thus we should take a look at how object mode works in ShadowNode, if that implementation takes too much memory or performance, I think we should reconsider this feature :)

**Comment by Txiaozhe** [comment]
I have read some source code about stream module of ShadowNode, but I didn't find anything special with node except it is not perfect yet. If we keep it's data struct in array, I think it is not very appropriate to do some slow operate. And if there are few practical scenarios for `objectNode` option, I think we could shelve it temporary!

**Comment by yorkie** [comment]
Array is simpler than linked one, the linked object should take more memory.

**Comment by Txiaozhe** [comment]
Yeah, stream of ShadowNode is simpler and more resource-efficient, so if there are few practical scenarios for `objectNode` option, I think we could shelve it temporary
